classes/publication/Publication.inc.php

```diff
@@ -15,6 +15,30 @@
  */
 import('lib.pkp.classes.publication.PKPPublication');
 
-class Publication extends PKPPublication {}
+class Publication extends PKPPublication {
+
+	/**
+	 * Get the URL to a localized cover image
+	 *
+	 * @param int $contextId
+	 * @return string
+	 */
+	public function getLocalizedCoverImageUrl($contextId) {
+		$coverImage = $this->getLocalizedData('coverImage');
+
+		if (!$coverImage) {
+			return '';
+		}
+
+		import('classes.file.PublicFileManager');
+		$publicFileManager = new PublicFileManager();
+
+		return join('/', [
+			Application::get()->getRequest()->getBaseUrl(),
+			$publicFileManager->getContextFilesPath($contextId),
+			$coverImage['uploadName'],
```

Comment:
I grepped around and it looks to be the case, but just checking -- this is not something derived from the client-side filename, correct? If so, it would definitely need to be sanitized to avoid misuse in e.g. this `join` (and other places).

Reply:
No, calling it `uploadName` is really just to keep it consistent with how our `FieldUploadImage` components work, and makes image objects consistent across cover images and context images.

Eventually we'll do away with this, I hope, in line with our comments here: https://github.com/pkp/pkp-lib/pull/5102#discussion_r331721815